File: src/main/java/io/github/mdsimmo/bomberman/commands/game/set/Set.java
package io.github.mdsimmo.bomberman.commands.game.set;

import io.github.mdsimmo.bomberman.commands.Cmd;
import io.github.mdsimmo.bomberman.commands.CommandGroup;
import io.github.mdsimmo.bomberman.messaging.Message;
import io.github.mdsimmo.bomberman.messaging.Text;

public class Set extends CommandGroup {

    public Set(Cmd parent) {
        super(parent);
        addChildren(
                new Schema(this),
                new Lives(this),
                new MinPlayers(this)
        );
    }

    @Override
    public Message name() {
        return context(Text.SET_NAME).format();
    }

    @Override
    public Permission permission() {
        return Permission.GAME_DICTATE;
    }

    @Override
    public Message description() {
        return context(Text.SET_DESCRIPTION).format();
    }

}

File: src/main/java/io/github/mdsimmo/bomberman/commands/game/set/Set.kt
package io.github.mdsimmo.bomberman.commands.game.set

import io.github.mdsimmo.bomberman.commands.Cmd
import io.github.mdsimmo.bomberman.commands.CommandGroup
import io.github.mdsimmo.bomberman.messaging.Message
import io.github.mdsimmo.bomberman.messaging.Text

class Set(parent: Cmd) : CommandGroup(parent) {
    override fun name(): Message {
        return context(Text.SET_NAME).format()
    }

    override fun permission(): Permission {
        return Permission.GAME_DICTATE
    }

    override fun description(): Message {
        return context(Text.SET_DESCRIPTION).format()
    }

    init {
        addChildren( //new Schema(this),
                SetLives(this)
        )
    }
}